File: whsk2/Colour.cpp
#include "Colour.h"

using Engine::Colour;

Colour Colour::red = Colour(1, 0, 0, 1), Colour::green = Colour(0, 1, 0, 1), Colour::black = Colour(0, 0, 0, 1), Colour::blue = Colour(0, 0, 1, 1), Colour::yellow = Colour(1, 1, 0, 1), Colour::orange = Colour(1, 0.5f, 0, 1), Colour::white = Colour(1, 1, 1, 1), Colour::purple = Colour(0.5f, 0, 0.5f, 1), Colour::pink = Colour(1,0.8f,0.8f,1), Colour::grey = Colour(0.5f,0.5f,0.5f,1), Colour::brown = Colour(0.5f, 0.25f, 0, 1);

Colour::Colour(float r, float g, float b, float a)
	:
	r(r),
	g(g),
	b(b),
	a(a)
{

}

Colour::Colour()
	:
	Colour(1,1,1,1)
{

}

Colour::~Colour()
{

}

File: whsk2/lua/corelibs/Colour.lua
Colour = {}
Colour.__index = Colour

function Colour.new(r,g,b,a)

	local colour = {}
	setmetatable(colour, Colour)
	
	colour.r = r
	colour.g = g
	colour.b = b
	colour.a = a

	return colour

end

Colour.white = Colour.new(1,1,1)
Colour.red = Colour.new(1,0,0,1)
Colour.green = Colour.new(0, 1, 0, 1)
Colour.black = Colour.new(0, 0, 0, 0)
Colour.blue = Colour.new(0, 0, 1, 1)
Colour.yellow = Colour.new(1, 1, 0, 1)
Colour.orange = Colour.new(1, 0.5, 0, 1)
Colour.pink = Colour.new(1, 0.8, 0.8, 1)
Colour.purple = Colour.new(0.5, 0, 0.5, 1)
Colour.grey = Colour.new(0.5, 0.5, 0.5, 1)
Colour.brown = Colour.new(0.5, 0.25, 0, 1)

return Colour